scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -716,8 +718,10 @@ private void processApplicationDataRecord(final Record record) {
 						// thus, the handshake seems to have been completed successfully
 						connection.handshakeCompleted(record.getPeerAddress());
 						session.markRecordAsRead(record.getEpoch(), record.getSequenceNumber());
-						// finally, forward de-crypted message to application layer
-						handleApplicationMessage(message, session);
+						// create application message, if handler is set.
+						if (messageHandler != null) {
```

Comment:
Should we keep this test ?

Reply:
If agreed, that such a connector without messageHandler doesn't make sense, we can remove it and just keep the check below.

Reply:
Ok except, that sounds good to merge.

Reply:
so remove the "if"?

Reply:
yes.